### PR TITLE
docker and github actions

### PR DESCRIPTION
**Pull request by mathematicalmichael**
First, thanks for this repo! I really enjoyed your youtube video and was excited at the opportunity to play around with the code.

I am hoping to contribute a helpful compilation option, and also set up some (free) CI testing courtesy of Github (the actions file).

I was able to simulate a couple dozen generations, cut the program, and then still make some movie files, so it seems to be fully working as far as I can tell.

Got it working with `docker` on both an intel-based x86 home server and just for kicks, a raspberry pi 4 as well. The ubuntu base image specified is architecture-aware. verified that it compiles on an M1 mac as well with this same Dockerfile. Took some time to build, especially on the ARM machines, so I uploaded some images:

- `mindthegrow/biosim4:latest` (should support multi-arch)
- `mindthegrow/biosim4:latest-amd64` / `mindthegrow/biosim4:latest-aarch64`

changes are made to support this execution pattern (paths relative to root directory of repo), I wanted to have the README instructions not yield any warnings/errors.

```
docker build -t biosim4
docker run --rm -ti -v `pwd`:/app --name biosim biosim4 make
docker run --rm -ti -v `pwd`:/app --name biosim biosim4 ./bin/Release/biosim4 biosim4.ini
```

`dumb-init` needed to be added to handle `Ctrl-c` for the command above which invokes the executable.
I noticed I got the `sensorVal` message repeatedly on amd64 but not arm64. I had to comment it out in order to get a readable stdout. 

If I added any packages that are not actually required, please do let me know. I like to keep images slim in size but the priority here was just getting something that worked out the door. I suppose `mpi` is missing and it does occur to me you mentioned you did get that working in your narration, but ... one step at a time =) I figure if people land on the repo, this provides a relatively low-friction entry point to reproducing the work.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -275,6 +294,16 @@ if the parameter named "displaySampleGenomes" is set to nonzero in the config fi
...
 graph-nnet.py.
 
+
+Note: If using the `docker run ... bash` command, the presumed directory structure would necessitate the
+following syntax:
+
+```sh
+cd tools;
+gnuplot graphlog.gp
+python3 graph-nnet.py
```
Comment (mathematicalmichael) [inline README.md:304]:
```suggestion
gnuplot tools/graphlog.gp
python3 tools/graph-nnet.py
```

is the latter correct?

File: README.md [hunk 2]
```diff
@@ -255,6 +270,10 @@ command line argument, e.g.:
 ./bin/Release/biosim4 [biosim4.ini]
 ```
 
+Note: For `docker`, `docker run --rm -ti -v `pwd`:/app --name biosim biosim4 bash` will put you into
```
Comment (mathematicalmichael) [inline README.md:273]:
```suggestion
Note: For `docker`, 
```sh
docker run --rm -ti -v `pwd`:/app --name biosim biosim4 bash
```
will put you into
```

**Review by mathematicalmichael — COMMENTED** [review]
readme formatting

File: README.md [hunk 3]
```diff
@@ -213,14 +213,29 @@ cimg-dev is replaced with version 2.8.4 or later.
...
 * The file named "biosim4.cbp" is a configuration file for the Code::Blocks IDE version 20.03.
 
 * A Makefile is provided which
 was created from biosim4.cbp with cbp2make, but is not tested. A default "make" will generate a debug and a
 release version.
 
+
+* A Dockerfile is provided which leverages the aforementioned Makefile.
+
+To compile the program:
```
Comment (mathematicalmichael) [inline README.md:227]:
```suggestion
To build the environment in which you can compile the program:
```

**Comment by davidrmiller** [comment]
@mathematicalmichael , this is impressive work. Thanks for making the images available.

**Comment by mathematicalmichael** [comment]
@davidrmiller what's impressive was your simulations. I just wanted to run them conveniently. Today I got to actually dig in to the config =)

my changes appear to be settled, do you have anything you'd like me to change to get this merged?

**Pull request merged**